homeassistant/components/squeezebox/media_player.py

```diff
@@ -143,12 +178,35 @@ async def _player_discovered(player: SqueezeBoxPlayerUpdateCoordinator) -> None:
         {
             vol.Required(ATTR_COMMAND): cv.string,
             vol.Optional(ATTR_PARAMETERS): vol.All(
-                cv.ensure_list, vol.Length(min=1), [cv.string]
+                cv.ensure_list,
+                vol.Length(min=1),
+                [cv.string],
             ),
         },
-        "async_call_query",
+        async_call_query_helper,
+        supports_response=SupportsResponse.OPTIONAL,
```

Comment:
OK, so slightly long answer...  Currently, we have 2 general services, call_query and call_method which can be used to call any function in the squeezebox api either with a response or without.  The existing process for returning data when using call_query is to set a state attribute rather than using the response_variable approach, I presume because this was done long before the response_variable approach existed.  We've had request to add the response_variable approach for call_query and had an earlier PR on this.  However, joostelek wouldn't approve that PR until we also made it easier for people to perform common actions (e.g. like search and play) without detailed knowledge of the API.  Whilst implementing this new simpler services, it doesn't make sense to have a different return approach to call_query as that would be confusing.  However, it also doesn't seem to make sense to introduce these new services, where people will either adapt or build new scripts/automations etc. using the attribute approach, which we'd like to migrate away from over time.  However, we also can't remove that approach short term as it's quite widely used and would be a nasty breaking change.  So, the approach here is to modify call_query to allow it to support both the old attribute approach or, if response_variable is specified, then to use the modern return approach.  Also, this means that people using these new services are more likely to use the modern approach, but they can also use the attribute  approach if that's easier in the short term.